### PR TITLE
feat: added driver configuration

### PR DESCRIPTION
**Pull request by shanduur**
Added configuration for the driver. This allows forcing behavior in the driver to further test sidecar and controller. It also includes basic implementation of real S3 client, that can perform bucket creation, but is unable to dynamically provision accesses.

### CODE REVIEW

File: pkg/clients/common.go [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package clients
+
+import (
+	"context"
+
+	cosi "sigs.k8s.io/container-object-storage-interface-spec"
```
Comment (BlaineEXE) [inline pkg/clients/common.go:19]: Should this be attempting to use new monorepo?

```suggestion
	cosi "sigs.k8s.io/container-object-storage-interface/proto"
```

File: go.mod [hunk 2]
```diff
@@ -1,37 +1,38 @@
...
+	github.com/minio/minio-go/v7 v7.0.82
 	github.com/pkg/errors v0.9.1
-	github.com/spf13/cobra v1.7.0
-	github.com/spf13/pflag v1.0.5
-	github.com/spf13/viper v1.15.0
+	github.com/stretchr/testify v1.10.0
 	google.golang.org/grpc v1.55.0
 	k8s.io/klog/v2 v2.100.1
 	sigs.k8s.io/container-object-storage-interface-provisioner-sidecar v0.1.0
 	sigs.k8s.io/container-object-storage-interface-spec v0.1.0
```
Comment (BlaineEXE) [inline go.mod:12]: Is it possible to migrate these to the monorepo?

**Review by BlaineEXE — COMMENTED** [review]
At this point, I think velocity is a bit more important than perfect code. To that end, I only gave this a quick overview.

Having the ability to return fake errors seems very needed, and from the `config/default/config.yaml` I think I understand how that works.

I think it makes sense for us to be using the monorepo here, but it's also fine to change that in a follow-up PR if needed.

File: pkg/clients/common.go (hunk 1 above)
Comment (shanduur) [reply pkg/clients/common.go:19]: I'll fix that in future, we need to fix the import paths in the core repo first.

**Pull request merged**